v1/lib/server/readMetadata.js

```diff
@@ -37,6 +38,8 @@ const SupportedHeaderFields = new Set([
   'custom_edit_url',
 ]);
 
+program.option('--skip-next-release').parse(process.argv);
```

Comment:
Can you move this to `generate.js` ? I think it's better to have all the cli args to be parsed on the actual cli script like below

https://github.com/facebook/Docusaurus/blob/a062ef33b684d57834eb0d57e0672a700cc75a6e/v1/lib/server/generate.js#L40

Reply:
I agree with you and I have already tried to put it in `generate.js`. The problem is that `generate.js` requires `sitemap.js` which calls `readMetadata.generateMetadataDocs()` when loaded as below
https://github.com/facebook/Docusaurus/blob/6b1d2e8c9ceeb1b8ec4728fdd51800d237661490/v1/lib/server/sitemap.js#L21-L23
In addition to this, `generate.js` also requires `core/site.js` which requires `nav/HeaderNav.js` and `HeaderNav.js` also calls `readMetadata.generateMetadataDocs()` when loaded as below
https://github.com/facebook/Docusaurus/blob/6b1d2e8c9ceeb1b8ec4728fdd51800d237661490/v1/lib/core/nav/HeaderNav.js#L23-L25
Hence, in one execution of `yarn build`, `readMetadata.generateMetadataDocs()` is called multiple times at multiple places. If we parse and check command options only in `generate.js` something will go wrong since other parts of the system will generate a complete metadata without excluding `next` release. Because of this reason, I think it is appropriate to do command parsing and checking at the point of metadata generation i.e. `readMetadata`.

Does this sound like a good reason to put it in `readMetadata`?

Reply:
Thanks for the explanation. Really appreciate it.

> readMetadata.generateMetadataDocs() is called multiple times at multiple places

Oh god, this is such a tech debt.  I think I've encountered this too previously when I tried to refactor things out. It's fortunate that we fixed this with a rewrite on next major version 😓 .